sdk/dynamicconfig/sdk/metric/controller/notifier/config.go

```diff
@@ -42,10 +48,6 @@ func GetDefaultConfig(period pb.ConfigResponse_MetricConfig_Schedule_CollectionP
 }
 
 func (config *Config) Validate() error {
```

Comment:
It may be worth checking that `MetricConfig` (or any field that you use, for that matter) is not `nil`. Since protobuf3, required fields are no longer a thing, so there's no guarantee that a config response has any fields at all. Indeed, `MetricConfig` will be `nil` in the event that a response consists only of a `Fingerprint`, when there are no new metric changes. What are your thoughts?

Reply:
So I was thinking about this, and I replaced this with `MetricConfigValidate`. I think it's definitely important to make sure it's not a nil, but if we only intend to configure tracing, I don't think having `MetricConfig` as nil is strictly "wrong".